src/websocket/atomic-swap.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import BaseWebSocket, { WsOptions } from './base';
+
+/**
+ * Handles websocket connections and message transmission
+ *
+ * This class extends the base websocket class.
+ *
+ * @class
+ * @name AtomicSwapWebSocket
+ */
+class AtomicSwapWebSocket extends BaseWebSocket {
```

Comment:
Is there any difference between this class and `WalletWebSocket`? The only one I've seen is the type split, which could be a parameter.

Reply:
Good point! I refactored all my code to use the `WalletWebSocket` instead, even the tests.
And since this class no longer serves only the wallet, I renamed it to `GenericWebSocket` on [bb5f334](https://github.com/HathorNetwork/hathor-wallet-lib/pull/533/commits/bb5f334dca068f64c250e8f41943d23ae2eb0795).

Do you agree with these changes?